### ISSUE TITLE
Winget does nothing -- progress bar stays at 0%

### ISSUE THREAD

**Issue by xobs** [microsoft/winget-cli#3459]
### Brief description of your issue

Winget appears to do nothing. It gets stuck at 0% when attempting to do anything involving a progress bar.

![image](https://github.com/microsoft/winget-cli/assets/238325/df20b74c-c8a5-4ee4-816b-049ee21ccac4)

### Steps to reproduce

1. Run a command, such as `winget search winget` or `winget upgrade` or `winget export output.txt`

### Expected behavior

Something should happen

### Actual behavior

The progress bar stays at 0%.

In order to exit, I must press Ctrl-C twice.

### Environment

```shell
Windows Package Manager v1.4.11071
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.1817
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.11071.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by stephengillie** [comment]
Can you try uninstalling WinGet with the Windows Store, then reinstalling from https://aka.ms/getwinget ?

**Comment by xobs** [comment]
I can, but I'm filing an issue here because it's not working correctly, and it can be improved. The winget CLI isn't properly handling something, and is getting stuck in some sort of loop. My system is in a state where we can work to identify the source of the problem rather than papering over it.

I feel that it shouldn't get into a state where it doesn't do anything, which is why I have filed the bug. "Reinstall the program" is a deeply unsatisfactory solution, but if there isn't interest in solving the cause of these issues then I can simply try working around it by reinstalling.

**Comment by denelon** [comment]
We've done some work in WinGet 1.5 to catch a few of the edge cases where this happens. Can you try just upgrading to WinGet 1.5 and see if this problem persists?

**Comment by xobs** [comment]
How do I get WinGet 1.5? The link at https://aka.ms/getwinget installs Version 1.20.1881.0, not 1.50:

![image](https://github.com/microsoft/winget-cli/assets/238325/2c167527-3b4e-4e95-9e0c-3ef314542772)

**Comment by Trenly** [comment]
> How do I get WinGet 1.5? The link at https://aka.ms/getwinget installs Version 1.20.1881.0, not 1.50:
> 
> ![image](https://user-images.githubusercontent.com/238325/256066278-2c167527-3b4e-4e95-9e0c-3ef314542772.png)

The version 1.20.1881.0 of AppInstaller should be the 1.5 version of WinGet. The reason for the discrepancy is because WinGet is bundled inside of AppInstaller. You can verify the version of WinGet is installed after the upgrade using `winget --version`

**Comment by xobs** [comment]
Alright, that's a bit confuising then. Regardless, the installer is taking a long time. Here is how far it is after four hours:

![image](https://github.com/microsoft/winget-cli/assets/238325/a1dfd662-0994-4081-9877-328b09257f6f)

**Comment by amritashan** [comment]
Any update to this? I am reinstalling winget, and it is stuck at 10%.
Mine is v1.22.x

![image](https://github.com/microsoft/winget-cli/assets/11738683/64bc8c20-61ab-4225-b413-1149033693e2)

**Comment by denelon** [comment]
We've released WinGet 1.8.
Is this problem still happening?

**Comment by xobs** [comment]
My theory is that this was due to a failure in the Windows Installer service, which had zero feedback on the failure.

Because Winget didn't have a timeout, it would just sit there.

The fix would be to prevent the Windows Installer Service from breaking, and provide information on why that service failed. However, within the confines of Winget, the workaround would be to detect that things are not right and put up an error.

As the original reporter, I am no longer experiencing this issue since I have since restarted the entire system and things are working again. I can't say whether Winget 1.8 fixes the issue since I can no longer reproduce it myself.
